pmlc/dialect/op_lib/cpp/utils.h

```diff
@@ -0,0 +1,98 @@
+//===- utils.cc - MLIR op lib dialect tblgen utils for C++ ----------------===//
+//
+// Copyright 2019 Intel Corporation.
+//
+// =============================================================================
+//
+// Tablegen utilities for the C++ outputs of the op lib dialect. This
+// encompasses things such as headers, includes, and type mappings.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <map>
+
+#include "mlir/Support/STLExtras.h"
+#include "llvm/ADT/StringExtras.h"
+
+using mlir::StringRef;
+
+namespace pmlc::dialect::op::tblgen::cpp {
+
+static inline const char* const fileCommentHeader = R"( // Copyright 2019 Intel Corporation.
+/*===- TableGen'erated file -------------------------------------*- C++ -*-===*\
+|*                                                                            *|
+|* Op Lib C++ EDSL Wrapper                                                    *|
+|*                                                                            *|
+|* Automatically generated file, do not edit!                                 *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+)";
+
+static inline const char* const includeHeader = R"(
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "plaidml2/edsl/edsl.h"
+#include "plaidml2/op/ffi.h"
+
+)";
+
+static inline const char* const commentHeader = R"(
+//===----------------------------------------------------------------------===//
+// {0} {1}
+//===----------------------------------------------------------------------===//
+
+)";
+
+static inline const char* const ffiFunction = R"(
+namespace details {
+
+inline edsl::Value op(const std::string& name, const edsl::Value& args) {
+  return edsl::Value(ffi::call<plaidml_expr*>(plaidml_op_make, name.c_str(), args.as_ptr()));
+}
+
+} // namespace details
+
+)";
```

Comment:
I'm thinking we should leave this kind of thing outside of the generated code, and instead have them be part of `ops.cc` (the file that includes `ops.h.inc`)